improver/nbhood/circular_kernel.py

```diff
@@ -32,28 +32,50 @@
 
 import iris
 import numpy as np
-import scipy.ndimage.filters
+from scipy.ndimage.filters import correlate
 
 from improver.constants import DEFAULT_PERCENTILES
 from improver.utilities.cube_checker import (
     check_cube_coordinates, find_dimension_coordinate_mismatch)
 from improver.utilities.spatial import (
-    check_if_grid_is_equal_area, convert_distance_into_number_of_grid_cells)
+    check_if_grid_is_equal_area, distance_to_number_of_grid_cells)
 
 # Maximum radius of the neighbourhood width in grid cells.
 MAX_RADIUS_IN_GRID_CELLS = 500
 
 
-def circular_kernel(fullranges, ranges, weighted_mode):
+def check_radius_against_distance(cube, radius):
+    """Check required distance isn't greater than the size of the domain.
+
+    Args:
+        cube (iris.cube.Cube):
+            The cube to check.
+        radius (float):
+            The radius, which cannot be bigger than the size of the domain.
+
+    """
+    axes = []
+    for axis in ['x', 'y']:
+        coord = cube.coord(axis=axis).copy()
+        coord.convert_units('metres')
+        axes.append((max(coord.points) - min(coord.points)))
+
+    max_allowed = np.sqrt(axes[0] ** 2 + axes[1] ** 2)
```

Comment:
Can you change this to (say) 0.33 times the domain diagonal, and update the docstring to say:
```
radius (float):
    The radius, which cannot be more than a third of the size of the domain
```
Can you then remove the `MAX_DISTANCE_IN_GRID_CELLS` constant and all uses from this module.

Reply:
Do you mean `max_allowed = np.sqrt(axes[0] ** 2 + axes[1] ** 2) * 0.33`?

Reply:
Yep, that's what I was thinking.
This is a fairly arbitrary "sanity" limit to ensure we're not applying neighbourhood processing in a way that doesn't make sense contextually.

Reply:
Well it breaks 13 tests, so I may need some help looking through which and why

Reply:
from psychrometics to wind

Reply:
With * 0.5 there are only 2 tests failing.

Reply:
That seems fixable.  Let's stick with 0.5 then, and see what we can do about those unit tests.

Reply:
One has been fixed, and changes puhed, just wind direction left.

Reply:
I've now fixed the wind direction unit test.  The fix is a bit of a hack, but it demonstrates the required functionality is retained.  Wind direction unit tests are one of the few remaining to be upgraded (see https://github.com/metoppv/improver/issues/742 ), so this can be properly dealt with when we address the overall inputs issue.